[export] Fix torchbind constant folding

yiming0416 opened the pull request:
Summary: `CallTorchBind` should not be folded during constant folding

Test Plan:
```
buck2 run mode/dev-nosan sigmoid/inference/test:test_passes -- -r test_const_folding_torchbind
```

Reviewed By: henryoier

Differential Revision: D67721272




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov @BoyuanFeng

zhxchen17 commented on torch/_inductor/constant_folding.py:
```diff
@@ -220,6 +220,11 @@ def set_env(arg: torch.fx.Node) -> None:
             return self.unknown_value
 
+        if node.op == "call_function" and isinstance(
+            node.target, torch._higher_order_ops.torchbind.CallTorchBind
```
I don't think we wanna specialize on call_torchbind here, in fact, we probably should exclude all HOOs here to be safe.

yiming0416 committed:
[export] Fix torchbind constant folding (#144684)

Summary:

Ban HOPs from being folded during constant folding

Test Plan:
```
buck2 run mode/dev-nosan sigmoid/inference/test:test_passes -- -r test_const_folding_torchbind
```

Reviewed By: henryoier, zhxchen17

Differential Revision: D67721272
